### ISSUE TITLE
Cannot use ag-grid with ts-loader >= 0.9.3

### ISSUE THREAD

**Issue by Wykks** [TypeStrong/ts-loader#388]
When using ag-grid and ts-loader >= 0.9.3 with allowJs set to true, I get lots of error like this : 
```
Module '"[...]/node_modules/ag-grid/main"' has no exported member '[...]'.
```
It's fine with ts-loader 0.9.2

Maybe it's related to this issue: https://github.com/TypeStrong/ts-loader/issues/250
Since ag-grid use the typings field as well https://github.com/ceolter/ag-grid/blob/master/package.json#L9

**Comment by johnnyreilly** [comment]
Since 0.9.3 is when we added support for `allowJs` I would guess it's related to that.  Have you tried setting `allowJs` to `false` in your `tsconfig.json`?

**Comment by Wykks** [comment]
Unfortunately, I can't do that, because my project is too big, and rely on this option for now.

**Comment by johnnyreilly** [comment]
Do you want to try something for me?  If you tweak [this line](https://github.com/TypeStrong/ts-loader/blob/5359d6334746e8374c8f559d0d814ef4b604ae82/src/instances.ts#L92):

from:

```ts
    // if allowJs is set then we should accept js(x) files
    const scriptRegex = configFile.config.compilerOptions.allowJs
        ? /\.tsx?$|\.jsx?$/i
        : /\.tsx?$/i;
```

to: 

```ts
    const scriptRegex = /\.tsx?$/i;
```

does it resolve your issue?

**Comment by Wykks** [comment]
Yes, it works fine with this change (using v1.2.2)

**Comment by johnnyreilly** [comment]
OK.  So that's a little frustrating.  Essentially, when `allowJs` is set to true we allow both both `js(x)` and `ts(x)` files to be processed. For most people this works in the way they would hope.  Not in your case though; I'm not sure why.

I'd possibly be open to supporting this behaviour in some way in ts-loader but there's a couple of things we'd need first:

1. A **simple** repro of the issue.  That repro should be turned into an integration test that becomes part of our testpack.  With the tweak in place the test passes, without it, it fails.  
2. We then need to think of how this is wisely implemented.  It would be possible to just stick this behind a flag and have done but I'm not keen.  I want to understand what the issue is.  I'd really rather have `allowJs` just work.  If we need to add an extra flag to cater for this we need to have a good understanding of why and, crucially, a clear name for it so other people picking over the issue will "fall into the pit of success".

Over to you!

**Comment by johnnyreilly** [comment]
Related to https://github.com/TypeStrong/ts-loader/issues/316 cc @dschnare

**Comment by Wykks** [comment]
Here you go https://github.com/TypeStrong/ts-loader/pull/396
Probably not clean (I do not fully understand your test system), but it does reproduce the issue

**Comment by johnnyreilly** [comment]
I'm waiting for input from @pqr but it seems that the change introduced in 0.9.3 may have been problematic. The behaviour introduced there (details here https://github.com/TypeStrong/ts-loader/issues/316 ) effectively caters for an edge case: when the entry point is a js file. Because this is an edge case and also because it seems to be causing problems for others I'm considering putting the new behaviour behind a flag. That will make it still available for @dschnare and others that rely on it. I want a little more data before I commit to this but I wanted to let you know what I was thinking.

**Comment by johnnyreilly** [comment]
Should be fixed with 1.3.0
